### ISSUE TITLE
Non-elementwise or positional score_mod

### ISSUE THREAD

**Issue by AmoghDabholkar** [pytorch-labs/attention-gym#111]
If I were to use a non-elementwise score_mod like torch.softmax, would that make sense? I tried to generate the Triton kernel for this as follows:

```
def softmax_score_mod(score, b, h, q_idx, kv_idx):
    return torch.softmax(score, dim=-1)

test_mask(softmax_score_mod, mask_mod=causal_mask, skip_correctness=True)
```
 The score modification part of the generated Triton kernel tries to do something like a Softmax but fails. The real question I have is what are some hard constraints on what the score_mod function cannot be?

![Image](https://github.com/user-attachments/assets/a4d26938-4be4-406b-9768-a5ac80ecda21)

**Comment by Chillee** [comment]
@AmoghDabholkar It's "whatever we can fuse into one kernel" haha. To be clear, in this case your operation doesn't really make sense, since `score` is a scalar tensor.

Generally speaking, the operations that work are more or less pointwise operations + indexing. I'd say that you'd generally need to do contortions in order to use any other operations, since all the inputs are scalars.
